src/controller/controller.ts

```diff
@@ -0,0 +1,21 @@
+import {Listener} from "../event-manager/interfaces";
+
+export class Controller {
```

Comment:
I prefer to keep event-related staff in a separate class, for example:

```typescript
export interface Listener {
  update: (payload: any) => void;
}

export class EventManager {
  private observers = new Map<string, Listener[]>();

  subscribe(eventType: string, observer: Listener) {
    if (!this.observers.has(eventType)) {
      this.observers.set(eventType, []);
    }
    const observers = this.observers.get(eventType);
    observers.push(observer);
    this.observers.set(eventType, observers)
  }

  unsubscribe(eventType: string, observer) {
    if (this.observers.has(eventType)) {
      let removeIndex =  this.observers.get(eventType).indexOf(observer);
      this.observers.get(eventType).splice(removeIndex, 1);
    }
  }

  notify(eventType: string, payload: any) {
    if (this.observers.has(eventType)) {
      let observers = this.observers.get(eventType);
      observers.forEach(observer => observer.update(payload))
    }
  }
}
```

In JS it's possible to use callbacks instead of `Listener` so you can just replace it with callback functions.

Please note, the code above wasn't tested